src/FormBuilder/resources/js/form-builder/src/settings/group-donation-confirmation/template-tags/index.tsx

```diff
@@ -0,0 +1,62 @@
+import {__} from '@wordpress/i18n';
+import {useCopyToClipboard} from '@wordpress/compose';
+import {getFormBuilderWindowData} from '@givewp/form-builder/common/getWindowData';
+import {Ref, useState} from 'react';
+import {Button, PanelRow} from '@wordpress/components';
+import {copy as copyIcon} from '@wordpress/icons';
+
+import './styles.scss';
+
+const {donationConfirmationTemplateTags} = getFormBuilderWindowData();
+
+/**
+ * @unreleased
+ */
+function CopyTagButton({textToCopy}) {
+    const [isCopied, setCopied] = useState(false);
+    const ref = useCopyToClipboard(textToCopy, () => {
+        setCopied(true);
+
+        return setTimeout(() => setCopied(false), 1000);
+    });
+
+    return (
+        <Button
+            className="givewp-popover-content-settings__copy-button"
+            isSmall
+            variant="tertiary"
+            ref={ref as Ref<HTMLAnchorElement>}
+            icon={copyIcon}
+        >
+            {isCopied ? __('Copied!', 'give') : __('Copy Tag', 'give')}
+        </Button>
+    );
+}
+
+const TemplateTags = () => {
+    return (
+        <PanelRow>
+            <ul className="givewp-donation-confirmation-settings__template-tags-list">
+                {donationConfirmationTemplateTags.map(({id, description}) => {
+                    const tagId = `{${id}}`;
+
+                    return (
+                        <li className="givewp-donation-confirmation-settings__template-tag-list-item" key={id}>
+                            <div className="givewp-donation-confirmation-settings__template-tag-list-item-top">
+                                <span className="givewp-donation-confirmation-settings__template-tag">{tagId}</span>
+                                <CopyTagButton textToCopy={tagId} />
+                            </div>
+                            <div className="givewp-donation-confirmation-settings__template-tag-list-item-bottom">
+                                <span className="givewp-donation-confirmation-settings__template-description">
+                                    {description}
+                                </span>
+                            </div>
+                        </li>
+                    );
+                })}
+            </ul>
+        </PanelRow>
+    );
+};
```

Comment:
@pauloiankoski this is a very common setting that we'll want to re-use. 

 Previously we were using the shared component `PopoverContentWithTemplateTags` and I would really like to preserve that functionality for this new flow.  Can you please create a new component for template tags that is not setting specific (similar to the previous component) and refactor to use that instead?  That way all settings that have template tags can use this and we can eventually move it into the form-builder-library for add-ons to use as well.

Reply:
@jonwaldstein I created a shared `TemplateTags` component (9b0182ddc8e46bb0bf0d85fdf60c9f7ed6a439fb) and refactored the `PopoverContentWithTemplateTags` component to use it (2ec138db3165a76eb15fb11c7d5632d51bcfcc39). The Donation Confirmation section now loads the `TemplateTags` component from the shared one.

In the future, let's move all these shared components into the `@givewp/form-builder-library`. From what I could see so far, Currency Switcher and Gift Aid are the add-ons that can benefit from this move.

Reply:
@pauloiankoski muito obrigado!!